pkgs/by-name/dw/dwl/package.nix

```diff
@@ -0,0 +1,125 @@
+{
+  lib,
+  fetchFromGitea,
+  installShellFiles,
+  libX11,
+  libinput,
+  libxcb,
+  libxkbcommon,
+  pixman,
+  pkg-config,
+  stdenv,
+  testers,
+  wayland,
+  wayland-protocols,
+  wayland-scanner,
+  wlroots,
+  writeText,
+  xcbutilwm,
+  xwayland,
+  # Boolean flags
+  enableXWayland ? true,
+  withCustomConfigDefH ? (configDefH != null || conf != null),
+  # Configurable options
+  configDefH ?
+    if conf != null then lib.warn "conf parameter is deprecated; use configDefH instead" conf else null,
+  conf ? null,
+}:
+
+# If we set withCustomConfigDefH, let's not forget configDefH
+assert withCustomConfigDefH -> (configDefH != null || conf != null);
+stdenv.mkDerivation (finalAttrs: {
+  pname = "dwl";
+  version = "0.6";
+
+  src = fetchFromGitea {
+    domain = "codeberg.org";
+    owner = "dwl";
+    repo = "dwl";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-fygUzEi4bgopesvHByfpatkLFYI98qozJOUBNM2t9Mg=";
+  };
+
+  nativeBuildInputs = [
+    installShellFiles
+    pkg-config
+    wayland-scanner
+  ];
+
+  buildInputs =
+    [
+      libinput
+      libxcb
+      libxkbcommon
+      pixman
+      wayland
+      wayland-protocols
+      wlroots
+    ]
+    ++ lib.optionals enableXWayland [
+      libX11
+      xcbutilwm
+      xwayland
+    ];
+
+  outputs = [
+    "out"
+    "man"
+  ];
+
+  postPatch =
+    let
+      configFile =
+        if lib.isDerivation configDefH || builtins.isPath configDefH then
+          configDefH
+        else
+          writeText "config.def.h" configDefH;
+    in
+    lib.optionalString withCustomConfigDefH "cp ${configFile} config.def.h";
```

Comment:
What is the reason for this change away from conf argument? Wouldn't it make more sense to keep only conf, but copy it to `config.h` instead of `config.def.h`? To really override the config every time, not only if the source doesn't have config.h.

Reply:
I am not understanding you here.
I have changed the `conf` argument to the more descriptive `configDefH` (now configH).
Your suggestion is then to move this configuration directly at `config.h`, instead of relying on the upstream build scripts that will move `config.def.h`? Well, it looks OK to me.

Reply:
It's not about relying on those scripts. It's about this: imagine you have a `config.h` in your repository. Then you fetch it by overriding `src` of dwl in nixpkgs, and set `configH`, expecting you will get the config you set in `configH`. But you won't, because it goes to `config.def.h` that is copied only if `config.h` doesn't exist.

I don't mind much if one or the other way is used. On the other hand it looks very good to me to either choose `configDefH` (or `configDefaultH` to be more clear) to override `config.def.h`, or `configH` to override `config.h`. How it is right now, called `configH`, but setting `config.def.h` is confusing to me, the same as `conf` setting `config.def.h` was confusing to me as well.

Reply:
I never thought this way. I will do this.